### PR TITLE
feat(@deepnote/cli): add convert command for notebook format conversion

### PR DESCRIPTION
**Pull request by jamesbhobbs**
Integrates @deepnote/convert into the CLI with support for:
- Converting Jupyter, Quarto, percent, and Marimo formats to Deepnote
- Converting Deepnote to Jupyter, Quarto, percent, or Marimo formats
- Directory conversion, custom output paths, and JSON output

Also fixes naming inconsistency in @deepnote/convert by renaming convertDeepnoteFileToJupyter to convertDeepnoteFileToJupyterFiles for consistency with other export functions.

## Summary by CodeRabbit

* **New Features**
  * Added a `convert` CLI command to convert between Deepnote, Jupyter, Quarto, Percent, and Marimo with file/directory workflows, naming/output options, JSON/quiet output, and progress feedback.
* **Tests**
  * Extensive test coverage for conversions, round-trips, directory handling, edge cases, and many new format fixtures.
* **Refactor**
  * Unified Jupyter conversion API surface.
* **Chores**
  * Added CLI dependencies to support conversion and spinner UX.
* **Documentation**
  * README examples updated for the conversion API.
* **UX**
  * Shell completions updated to include the new convert command.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/cli/src/commands/convert.test.ts [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+
+app = marimo.App()
+
+@app.cell
+def __():
+    x = 1
+    return x,
+
+if __name__ == "__main__":
+    app.run()
```
Comment (dinohamzic) [inline packages/cli/src/commands/convert.test.ts:112]: It would be nice to have example files like we have in /examples and reuse across packages. Or do you think this would be overkill?

File: packages/cli/src/commands/convert.ts [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+  const absolutePath = resolve(cwd, inputPath)
+
+  // Check if path exists
+  const stat = await fs.stat(absolutePath).catch(() => null)
+  if (!stat) {
+    throw new Error(`File or directory not found: ${inputPath}`)
+  }
+
+  const isDirectory = stat.isDirectory()
+  const ext = isDirectory ? '' : extname(absolutePath).toLowerCase()
```
Comment (dinohamzic) [inline packages/cli/src/commands/convert.ts]: Should we extract file resolution logic into utils/file-resolver?

File: packages/cli/src/commands/convert.ts [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+    return convertFromDeepnote(absolutePath, outputFormat, options)
+  }
+  return convertToDeepnote(absolutePath, inputFormat, isDirectory, options)
+}
+
+async function determineInputFormat(
+  absolutePath: string,
+  isDirectory: boolean,
+  ext: string
+): Promise<'jupyter' | 'quarto' | 'percent' | 'marimo' | 'deepnote'> {
```
Comment (dinohamzic) [inline packages/cli/src/commands/convert.ts]: Could we cover this with types from @deepnote/convert?

File: packages/cli/src/cli.ts [hunk 4]
```diff
@@ -212,6 +213,60 @@ ${c.bold('Exit Codes:')}
...
+    .description('Convert between notebook formats (.ipynb, .qmd, .py, .deepnote)')
+    .argument('<path>', 'Path to a file or directory to convert')
+    .option('-o, --output <path>', 'Output path (file or directory)')
+    .option('-n, --name <name>', 'Project name (for conversions to .deepnote)')
+    .option(
+      '-f, --format <format>',
+      'Output format when converting from .deepnote (jupyter, percent, quarto, marimo)',
+      'jupyter'
+    )
+    .option('--json', 'Output result in JSON format for scripting')
```
Comment (dinohamzic) [inline packages/cli/src/cli.ts]: This is not a valid option anymore.

File: packages/cli/src/cli.ts [hunk 5]
```diff
@@ -212,6 +213,60 @@ ${c.bold('Exit Codes:')}
...
+  $ deepnote convert project.deepnote
+
+  ${c.dim('# Convert Deepnote to Quarto')}
+  $ deepnote convert project.deepnote -f quarto
+
+  ${c.dim('# Convert Deepnote to Marimo')}
+  $ deepnote convert project.deepnote -f marimo
+
+  ${c.dim('# Output result as JSON for CI/CD')}
+  $ deepnote convert notebook.ipynb --json
```
Comment (dinohamzic) [inline packages/cli/src/cli.ts]: Same here.

File: packages/cli/src/commands/convert.test.ts (hunk 1 above)
Comment (Artmann) [reply packages/cli/src/commands/convert.test.ts:112]: Yeah, that would be good, but as text fixtures and not as `examples/` as they are meant for users.

File: packages/cli/src/cli.ts (hunk 4 above)
Comment (Artmann) [reply packages/cli/src/cli.ts]: Fixed

File: packages/cli/src/cli.ts (hunk 5 above)
Comment (Artmann) [reply packages/cli/src/cli.ts]: Fixed

File: packages/cli/src/commands/convert.test.ts (hunk 1 above)
Comment (Artmann) [reply packages/cli/src/commands/convert.test.ts:112]: Created `test-fixtures/`

Comment (dinohamzic) [reply packages/cli/src/commands/convert.test.ts:112]: Amazing 🙏

File: packages/cli/src/commands/convert.ts (hunk 2 above)
Comment (Artmann) [reply packages/cli/src/commands/convert.ts]: Fixed

File: packages/cli/src/commands/convert.ts (hunk 3 above)
Comment (Artmann) [reply packages/cli/src/commands/convert.ts]: Fixed

File: test-fixtures/formats/marimo/dependencies.marimo.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+@app.cell
+def __(pd):
+    df = pd.read_csv("data.csv")
+    return df,
+
+@app.cell
+def __(df, np, pd):
+    result = df.mean()
+    return result,
```
[the remark this answers is not in the document]
Comment (Artmann) [reply test-fixtures/formats/marimo/dependencies.marimo.py:18]: This is just a test file

File: test-fixtures/formats/marimo/sql-cell.marimo.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+@app.cell
+def _(engine):
+    df = mo.sql(
+        f"""
+        SELECT * FROM film
+        """,
+        engine=engine
+    )
+    return (df,)
```
[the remark this answers is not in the document]
Comment (Artmann) [reply test-fixtures/formats/marimo/sql-cell.marimo.py:13]: This is just a test file
